spec/controllers/group_assignment_invitations_controller_spec.rb

```diff
@@ -97,6 +97,30 @@
           expect(user.repo_accesses.count).to eql(1)
         end
       end
+
+      context 'github repository with the same name already exists' do
+        before do
+          group = Group.create(title: 'The Group', grouping: grouping)
+          group_assignment_repo = GroupAssignmentRepo.create!(group_assignment: group_assignment, group: group)
+          @original_repository = organization.github_client.repository(group_assignment_repo.github_repo_id)
+          group_assignment_repo.delete
+          patch :accept_invitation, id: invitation.key, group: { id: group.id }
+        end
+
+        it 'creates a new group assignment repo' do
+          expect(group_assignment.group_assignment_repos.count).to eql(1)
+        end
+
+        it 'new repository name has expected suffix' do
+          expect(WebMock).to have_requested(:post, github_url("/organizations/#{organization.github_id}/repos"))
+            .with(body: /^.*#{@original_repository.name}-1.*$/)
+        end
+
+        after do
+          organization.github_client.delete_repository(@original_repository.id)
+          GroupAssignmentRepo.destroy_all
```

Comment:
you're going to want to add `Group.destroy_all` here as well so that the misc team that was created is removed from GitHub.